modules/api/pkg/ee/clusterbackup/storage-location/provider.go

```diff
@@ -26,77 +26,108 @@ package storagelocation
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apiserver/pkg/storage/names"
+	restclient "k8s.io/client-go/rest"
 	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 type BackupStorageProvider struct {
-	privilegedClient ctrlruntimeclient.Client
+	createMasterImpersonatedClient kubernetes.ImpersonationClient
+	privilegedClient               ctrlruntimeclient.Client
 }
 
 var _ provider.BackupStorageProvider = &BackupStorageProvider{}
 
-func NewBackupStorageProvider(privilegedClient ctrlruntimeclient.Client) *BackupStorageProvider {
+const (
+	ownerReferenceApi  = "kubermatic.k8s.io/v1"
+	ownerReferenceKind = "ClusterBackupStorageLocation"
+)
+
+func NewBackupStorageProvider(createMasterImpersonatedClient kubernetes.ImpersonationClient, privilegedClient ctrlruntimeclient.Client) *BackupStorageProvider {
 	return &BackupStorageProvider{
-		privilegedClient: privilegedClient,
+		createMasterImpersonatedClient: createMasterImpersonatedClient,
+		privilegedClient:               privilegedClient,
 	}
 }
 
-func (p *BackupStorageProvider) ListUnsecured(ctx context.Context, labelSet map[string]string) (*kubermaticv1.ClusterBackupStorageLocationList, error) {
+func (p *BackupStorageProvider) List(ctx context.Context, userInfo *provider.UserInfo, labelSet map[string]string) (*kubermaticv1.ClusterBackupStorageLocationList, error) {
+	if userInfo == nil {
+		return nil, errors.New("a user is missing but required")
+	}
+
 	listOpts := &ctrlruntimeclient.ListOptions{
 		LabelSelector: labels.SelectorFromSet(labelSet),
 		Namespace:     resources.KubermaticNamespace,
 	}
 
 	cbslList := &kubermaticv1.ClusterBackupStorageLocationList{}
-	if err := p.privilegedClient.List(ctx, cbslList, listOpts); err != nil {
+	client := p.privilegedClient
+
+	if err := client.List(ctx, cbslList, listOpts); err != nil {
 		return nil, fmt.Errorf("failed to list ClusterBackupStorageLocations: %w", err)
 	}
 	return cbslList, nil
 }
 
-// func (p *BackupStorageProvider) Get(ctx context.Context, userInfo *provider.UserInfo, name string) (*kubermaticv1.ClusterBackupStorageLocation, error).
-func (p *BackupStorageProvider) GetUnsecured(ctx context.Context, name string, labelSet map[string]string) (*kubermaticv1.ClusterBackupStorageLocation, error) {
-	// we use List() to filter CBSLs with labels easily, to keep scoped into the owner project.
-	cbslList, err := p.ListUnsecured(ctx, labelSet)
+func (p *BackupStorageProvider) Get(ctx context.Context, userInfo *provider.UserInfo, name string, labelSet map[string]string) (*kubermaticv1.ClusterBackupStorageLocation, error) {
+	if userInfo == nil {
+		return nil, errors.New("a user is missing but required")
+	}
+
+	cbslList, err := p.List(ctx, userInfo, labelSet)
 	if err != nil {
-		return nil, fmt.Errorf("failed to list ClusterBackupStorageLocations: %w", err)
+		return nil, err
 	}
 	for _, cbsl := range cbslList.Items {
 		if cbsl.Name == name {
 			return &cbsl, nil
```

Comment:
I know this is not part of this PR, but I would appreciate if we could create a ticket to investigate this. Judging from just the code I think it should be fine to create the impersonated client and run a client.Get. 

See https://github.com/kubermatic/dashboard/blob/03648c694cca330f62ad519cac4338a2bd9af6f5/modules/api/pkg/provider/kubernetes/project.go#L128-L135 for an example